doc/overlays.xml

```diff
@@ -62,37 +59,40 @@ the overlay, and add a symbolic link to it in the
 <section xml:id="sec-overlays-layout">
 <title>Overlays Layout</title>
 
-<para>An overlay is a Nix expression, which is a function which accepts 2
-arguments.</para>
+<para>Overlays are expressed as Nix functions which accept 2 arguments and return a set of
+packages</para>
 
 <programlisting>
 self: super:
 
 {
-  foo = super.foo.override { ... };
-  bar = import ./pkgs/bar {
-    inherit (self) stdenv fetchurl;
-    inherit (self) ninja crawl dwarf-fortress;
+  boost = super.boost.override {
+    python = self.python3;
+  };
+  rr = super.callPackage ./pkgs/rr {
```

Comment:
That sounds reasonable, but can you explain how functions already capture `self` if they need to? E.g. if I have an overlay that adds two new packages:
```
self: super: {
    # Let's say all deps are already in super,
    # so I can see this working with super.callPackages or self.callPackages
    foo = super.callPackage {};
    # But this depends on foo, which isn't in super
    bar = super.callPackage {}; 
}
```
How does the second `super.callPackage` get a reference to `foo` to provide to `bar`?